Lib/test/support/__init__.py

```diff
@@ -520,6 +521,17 @@ def requires_subprocess():
     """Used for subprocess, os.spawn calls, fd inheritance"""
     return unittest.skipUnless(has_subprocess_support, "requires subprocess support")
 
+# Emscripten's socket emulation has limitation. WASI doesn't have sockets yet.
+has_socket_support = not is_emscripten and not is_wasi
+
+def requires_working_socket(*, module=False):
```

Comment:
Could you add a docstring pointing out that if `module=False` this can be used as a decorator for `has_socket_support`?